### PR TITLE
feat(username): add POST /api/username/release to burn a caller's own handle

### PR DESCRIPTION
**Pull request by mbradley**
Backend half of the opt-in "burn my @divine.video username when I delete my account" feature (divinevideo/divine-mobile#6126). The mobile PR that adds the toggle depends on this endpoint.

## What
Adds a public `POST /api/username/release` so a user can permanently give up their own `@divine.video` username. Until now the only way to burn a name was the admin path, so a departing user's handle kept resolving after everything else was gone.

- **Auth:** NIP-98, same as `/claim`; the pubkey comes from the signed header.
- **Ownership:** verifies the authed pubkey currently holds the exact active name in the body before acting. You can only burn your own name.
- **Effect:** `revokeUsername(canonical, burn:true)` (status `burned`, `recyclable=0`) + Fastly de-sync, so the name stops resolving at origin and edge and cannot be re-registered (squatting/impersonation protection).
- **Idempotent:** 200 no-op when the caller owns no active name (e.g. already burned); 403 if they own a different name; 401 on NIP-98 failure.

## Why burn, not revoke
Burn blocks re-registration; revoke would let a squatter re-claim a departed user's handle. Opt-in on the client, gated behind the delete-account confirmation.

## Tests
4 new endpoint tests (burn own name, 403 wrong owner, 200 no-op, 401 auth). Full suite green (350), `tsc --noEmit` clean.

## Deployment
Needs `npm run deploy` (wrangler) to reach production `names.divine.video`, the host the mobile client calls. Safe to deploy ahead of the mobile PR — the endpoint is additive and unused until the toggle ships.

### CODE REVIEW

File: src/routes/username.ts [hunk 1]
```diff
@@ -557,4 +558,91 @@ username.post('/claim', async (c) => {
...
+      return c.json({ ok: false, error: 'Invalid JSON body' }, 400)
+    }
+    const { name } = body as { name?: unknown }
+    if (typeof name !== 'string' || name.length === 0) {
+      return c.json({ ok: false, error: 'name is required' }, 400)
+    }
+
+    let usernameData: { display: string; canonical: string }
+    try {
+      usernameData = validateUsername(name)
```
Comment (dcadenas) [inline src/routes/username.ts]: Stop applying the claim-time character gate to names already stored at `src/routes/username.ts:594`. Keep trim, lowercase, and IDNA/punycode canonicalization, but skip the underscore and dot rejection for `/release` so the ownership lookup and burn use the same canonical key. Extract that canonicalization so `/claim` keeps rejecting invalid new names, and add tests for an underscore name, a Unicode name that canonicalizes to punycode, and a non-owner returning 403.

**Review by dcadenas — CHANGES_REQUESTED** [review]
The release endpoint rejects legacy usernames containing underscores before it checks ownership, so affected users cannot burn their own handles and those names continue resolving after account deletion. Please separate canonicalization from the claim-time character validation and add coverage for underscore and Unicode names while preserving claim rejection and ownership checks.

File: src/routes/username.ts (hunk 1 above)
Comment (realmeylisdev) [inline src/routes/username.ts]: **should-fix:** `validateUsername` runs on the requested name *before* the ownership check, but `GET /by-pubkey` serves legacy vine-import names unvalidated — the import kept dots and underscores (`scripts/import-vine-users.ts` charset `[a-z0-9._-]`), and `fix-vine-underscores.ts` deliberately skipped rows with claimed Keycast accounts. So mobile offers the burn toggle for e.g. `the_funny_vine`, and the release always 400s ("cannot contain… underscores") → the opted-in burn hard-blocks that user's entire account deletion (divinevideo/divine-mobile#6138). Since this route already loads the caller's row and one pubkey holds at most one active name, comparing the requested name against the stored canonical/display (case-insensitively) instead of re-deriving through claim-time validation fixes every legacy shape.

File: src/routes/username.ts [hunk 2]
```diff
@@ -557,4 +558,91 @@ username.post('/claim', async (c) => {
...
+username.post('/release', async (c) => {
+  try {
+    // Read raw body text first (needed for NIP-98 payload verification)
+    const bodyText = await c.req.text()
+
+    const url = new URL(c.req.url)
+    // Lowercase for parity with /claim; getUsernameByPubkey compares
+    // case-insensitively, so this is defensive normalization.
+    const pubkey = (
+      await verifyNip98Event(
```
Comment (realmeylisdev) [inline src/routes/username.ts:578]: **observation:** the ±60s `created_at` window is inherited here, but for vine-import users `/release` is their first-ever NIP-98 call (names were imported, never claimed), and the mobile client stamps `created_at` *before* a Keycast sign RPC that can take up to 30s — its token-refresh retry re-signs the same pre-stamped event. Device clock skew plus signer latency → systematic 401 → the mobile hard-block makes deletion un-completable until the user unchecks the toggle. Worth considering a slightly wider window for this endpoint, or keeping the timestamp message distinct so clients can surface an actionable error.

File: src/routes/username.test.ts [hunk 3]
```diff
@@ -1104,3 +1126,143 @@ describe('Public Name Reservation', () => {
     })
   })
 })
+
+describe('POST /release - burn own username', () => {
```
Comment (realmeylisdev) [inline src/routes/username.test.ts:1152]: **nit (tests):** this suite fully mocks `verifyNip98Event` and never asserts its call args, so the contract mobile signs against — full absolute URL from `url.toString()`, method `POST`, raw unmodified body text — is unpinned; a refactor that passes a path-only URL or re-serialized body would keep these green while breaking every real client. One `expect(verifyNip98Event).toHaveBeenCalledWith(expect.anything(), 'POST', 'http://localhost/api/username/release', bodyText)` locks it.

**Review by realmeylisdev — COMMENTED** [review]
Cross-checked against the mobile half (divinevideo/divine-mobile#6138), which hard-blocks the whole account deletion on any /release failure — details inline; the legacy-name 400 is the one that bites.

File: src/routes/username.ts (hunk 1 above)
Comment (mbradley) [reply src/routes/username.ts]: Done in f464118. /release no longer runs the claim-time character gate: it canonicalizes (trim, lowercase, IDNA/punycode) and compares the requested name against the caller's stored canonical/display, so the ownership lookup and the burn share one key. /claim still rejects invalid new names. Tests cover an underscore legacy name, a Unicode name that canonicalizes to punycode, and a non-owner 403.

Comment (mbradley) [reply src/routes/username.ts]: Fixed in f464118, same approach: /release compares the requested name against the caller's stored canonical/display instead of re-deriving through claim-time validation, so every legacy vine-import shape resolves against its own stored key. `the_funny_vine` and a Unicode/punycode name both have burn tests now.

File: src/routes/username.ts (hunk 2 above)
Comment (mbradley) [reply src/routes/username.ts:578]: Good catch on the first-ever-NIP-98 timing for vine-import users. I widened this endpoint's window to 300s in f464118 to absorb the pre-sign timestamp plus a slow Keycast RPC and some clock skew, rather than keeping the inherited +/-60s. 300s is a judgment call: wide enough to stop the systematic 401 hard-block, but I'd like your read on whether that's too generous for an anti-replay window on a burn endpoint, or whether you'd rather keep 60s and have the client re-stamp after the sign. Happy either way.

File: src/routes/username.test.ts (hunk 3 above)
Comment (mbradley) [reply src/routes/username.test.ts:1152]: Pinned in f464118: the burn test now asserts verifyNip98Event was called with the exact absolute URL (http://localhost/api/username/release), POST, the raw unmodified body (JSON.stringify({name})), and the 300s window. A path-only URL or re-serialized body fails the test.

**Review by dcadenas — APPROVED** [review]
🔥 caller-owned handle release is ready
